subliminal/providers/legendastv.py

```diff
@@ -359,8 +375,16 @@ def query(self, language, title, season=None, episode=None, year=None):
                     if not a.pack and 'episode' in guess and guess['episode'] != episode:
                         continue
 
+                # calculates a good expiration time
+                if a.timestamp:
+                    delta = max(0, (datetime.datetime.now() - a.timestamp).total_seconds())
+                    # recently uploaded archives have short expiration time
+                    expiration_time = min(RELEASES_EXPIRATION_TIME, delta / 8)
+                else:
+                    expiration_time = RELEASES_EXPIRATION_TIME
+
                 # attempt to get the releases from the cache
-                releases = region.get(releases_key.format(archive_id=a.id))
+                releases = region.get(releases_key.format(archive_id=a.id), expiration_time=expiration_time)
```

Comment:
The only issue would be if you would like to manage the cache size or limit it to not go beyond certain value. I know that cli stores the cache on the disk. The python api leaves the decision to the client/developer whether to use disk or memory cache.